docs/reference/data-streams.asciidoc

```diff
@@ -0,0 +1,115 @@
+[chapter]
+[[data-streams]]
+= Data Streams
+
+You can use data streams to index time-based data that's continuously generated.
+A data stream groups indices from the same time-based data source.
+A data stream tracks its indices, known as _backing indices_, using an ordered
+list.
+
+A data stream's backing indices are <<index-hidden,hidden>>.
+The most recently created backing index is the
+write index. The other backing indices handle read requests. A data stream only
+accepts <<docs-index_,index requests>> with `op_type` set to `create`. To update
+or delete specific documents in a data stream, submit a <<docs-delete,delete>>
+or <<docs-update,update>> API request to the backing index containing the
+document.
+
+To create a data stream, set up a <<indices-templates,composable index
+template>> containing:
+
+* Any desired index patterns in the `index_patterns` property. Indices that
+  match these patterns are in the data stream's _namespace_.
```

Comment:
I find the mention of indices a little confusing and perhaps it's unfortunate that we're using the `index_patterns` parameter here since it's not really about indices. We don't really have (and I don't think we want to introduce) the concept of indices being in a data stream's namespace. While we do have a default naming convention for backing indices, we don't have any constraints on the names of indices that may be backing indices for a data stream, so we couldn't really define a "namespace" for a data stream. I think all that we want to communicate here is that when the pattern in the composable template matches the (not already existing) target of an index request, the corresponding data stream will be created.